### ISSUE TITLE
Ruby functions inside javascript (allowing AJAX requests to kinda work)

### ISSUE THREAD

**Issue by jphastings** [mynyml/harmony#12]
I'm trying to complete unit tests on some javascript I've written using [RightJS](http://rightjs.org) and their AJAX function [Xhr.load](http://rightjs.org/docs/xhr).

As far as I can tell envjs, Johnson (and hence Harmony) dont' yet support AJAX calls, so this won't work exactly as planned. However I should be able to hack around this if I can  do something like this:

```
class HarmonyHelpers
  def initialize(harmony_page)
    @harmony = harmony_page
  end

  def xhr_load(destination,args)
    # use mechanize etc to get and parse

    # somehow I'd need to set this.responseText with the response etc.
    @harmony.execute_js(args['onSuccess']) if it_works and args['onSuccess']
    @harmony.execute_js(args['onFailure']) if !itworks and args['onFailure']
  end
end
```

**tl;dr**: Is there a simple way for me to provide Ruby methods that can be called from the JS environment (specifically, overriding the ones in the JS environment)

Thanks in advance!

**Comment by jphastings** [comment]
I got carried away and made _something_ that works, how well it works I have no idea!

I'd appreciate your input! http://gist.github.com/558802
